src/statement.c

```diff
@@ -5060,6 +5061,101 @@ Statement *AsmStatement::syntaxCopy()
     return new AsmStatement(loc, tokens);
 }
 
+/************************ ExtStatement ***************************************/
+
+ExtAsmStatement::ExtAsmStatement(Loc loc, Expression *insn,
+                                 Expressions *args, Identifiers *names,
+                                 Expressions *constraints, int outputargs,
+                                 Expressions *clobbers, Identifiers *labels)
+        : Statement(loc)
+{
+    this->insn = insn;
+    this->args = args;
+    this->names = names;
+    this->constraints = constraints;
+    this->outputargs = outputargs;
+    this->clobbers = clobbers;
+    this->labels = labels;
+    this->gotos = NULL;
+}
+
+Statement *ExtAsmStatement::syntaxCopy()
+{
+    Expressions *c_args = Expression::arraySyntaxCopy(args);
+    Expressions *c_constraints = Expression::arraySyntaxCopy(constraints);
+    Expressions *c_clobbers = Expression::arraySyntaxCopy(clobbers);
+
+    return new ExtAsmStatement(loc, insn->syntaxCopy(), c_args, names,
+                               c_constraints, outputargs, c_clobbers, labels);
+}
+
+Statement *ExtAsmStatement::semantic(Scope *sc)
+{
+    // Fold the instruction template string.
+    insn = insn->semantic(sc);
+    insn->optimize(WANTvalue);
+
+    if (insn->op != TOKstring || ((StringExp *) insn)->sz != 1)
+        error("instruction template must be a constant char string");
```

Comment:
Also need an extra check below this:
```
if (labels && outputargs)
    error("extended asm statements with labels cannot have output constraints");
```
I'll send you a PR for all I've mentioned. ;)